File: Java/src/main/java/Solution201_300/Solution234.java
package Solution201_300;

import tool.ListNode;

/**
 * @author windows
 */
public class Solution234 {
    public boolean isPalindrome(ListNode head) {
        StringBuilder sb = new StringBuilder();
        while (head != null) {
            sb.append(head.val);
            head = head.next;
        }
        return sb.toString().equals(sb.reverse().toString());
    }


    public static void main(String[] args) {
        Solution234 solution234 = new Solution234();
    }
}

File: C++/src/Solution201_300/Solution234.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution234.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;


//Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};


bool isPalindrome(ListNode* head) {
    vector<int> l;
    while (head != nullptr){
      l.push_back(head->val);
      head = head->next;
    }
    for (int i = 0; i < l.size() / 2; ++i) {
      if (l[i] != l[l.size() - i - 1]){
        return false;
      }
    }
    return true;
}